api/checks/runs.go

```diff
@@ -15,15 +15,28 @@ import (
 
 func updateCheckRunSummary(ctx context.Context, summary summaries.Summary, suites ...shared.CheckSuite) (bool, error) {
 	log := shared.GetLogger(ctx)
+	if len(suites) < 1 {
+		return false, nil
+	}
 	product := summary.GetCheckState().Product
-
 	testRun := summary.GetCheckState().TestRun
+
+	// Attempt to update any existing check runs for this SHA.
+	checkRuns, err := getExistingCheckRuns(ctx, suites[0])
```

Comment:
Since all suites have access to `checks:read`, and this method loads all check runs for a given SHA, we can just call it once, for the first suite.

Reply:
Does `getExistingCheckRuns(suites[0])` give *all* check runs from all suites, or just the check runs for the first suite?

Reply:
ping @lukebjerring 

If not, it seems like you want to combine all check runs from all suites.

Reply:
It's getting _all_ CheckRun objects for a SHA, regardless of suite.